objectherkenning_openbare_ruimte/frame_extraction/frame_extraction.py

```diff
@@ -18,11 +21,96 @@
 settings = ObjectherkenningOpenbareRuimteSettings.get_settings()
 
 
+class FFMPEG_frame_extractor:
+    """
+    This class wraps around the ffmpeg command to facilitate frame extraction and
+    distortion correction from videos.
+
+    Checks are in place to ensure arguments are valid:
+    fps, cx, cy, k1, k2 must be float
+    input_file must be an existing file
+    output_folder must be an existing folder
+    log_file must point to a valid filename in an existing folder
+    """
+
+    def __init__(self, fps, distortion_settings):
+        self.cmd = None
+
+        cx = distortion_settings["cx"]
+        cy = distortion_settings["cy"]
+        k1 = distortion_settings["k1"]
+        k2 = distortion_settings["k2"]
+
+        if not isinstance(fps, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(cx, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(cy, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(k1, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(k2, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
```

Comment:
I would suggest using exceptions to increase robustness, something like

`if not all(isinstance(value, float) for value in [fps, cx, cy, k1, k2]):`
`            raise TypeError("All parameters must be floats.")`

Also, right now there are typos (e.g. the printing error for k2 says invalid type for fps and it prints the type of fps).

Reply:
Good point and neat one-liner for the checking! Will fix this